components/filter/filter-dimension-set-date-text-value.js

```diff
@@ -0,0 +1,149 @@
+import { getUTCDateTimeRange } from '../../helpers/dateTime.js';
+import { LitElement } from 'lit';
+import { LocalizeCoreElement } from '../../helpers/localize-core-element.js';
+
+/**
+ * A component to represent a possible date value that can be selected for a dimension set (the main filter dimension type) for predefined date ranges.
+ * A range property is used to define the preset text shown, as well as the start and end date values formatted as UTC strings.
+ * This component does not render anything, but instead gathers data needed for the d2l-filter.
+ */
+class FilterDimensionSetDateTextValue extends LocalizeCoreElement(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * Whether this value in the filter is disabled or not
+			 * @type {boolean}
+			 */
+			disabled: { type: Boolean, reflect: true },
+			/**
+			 * @ignore
+			 */
+			endValue: { type: String },
+			/**
+			 * REQUIRED: Unique key to represent this value in the dimension
+			 * @type {string}
+			 */
+			key: { type: String },
+			/**
+			 * @type {'today'|'lastHour'|'24hours'|'48hours'|'7days'|'14days'|'30days'|'6months'}
+			 */
+			range: { type: String },
+			/**
+			 * @ignore
+			 */
+			rangeNum: { type: Number, reflect: true },
+			/**
+			 * @ignore
+			 */
+			rangeType: { type: String, reflect: true },
+			/**
+			 * Whether this value in the filter is selected or not
+			 * @type {boolean}
+			 */
+			selected: { type: Boolean, reflect: true },
+			/**
+			 * @ignore
+			 */
+			startValue: { type: String },
+			/**
+			 * @ignore
+			 */
+			text: { type: String, reflect: true }
+		};
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this.selected = false;
+		this.text = '';
+		/** @ignore */
+		this.type = 'date';
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+
+		if (!this.range) return;
+
+		this._handleRange();
+
+		if (this.selected) {
+			// if the value is initially selected, startValue and endValue should be set in case used by consumer
+			const dateTimeRange = getUTCDateTimeRange(this.rangeType, this.rangeNum);
+			this.startValue = dateTimeRange.startValue;
+			this.endValue = dateTimeRange.endValue;
+		}
+	}
+
+	updated(changedProperties) {
```

Comment:
It seems like these computed properties could be set earlier in the life cycle with: 
```javascript
willUpdate(changedProperties) {
	...
}
```
I'm not sure if it really provides much benefit for filter though, given that filter does the actual rendering vs this component. Nevertheless, I'd opt to move those earlier if possible.